src/store/index.js

```diff
@@ -322,7 +330,7 @@ export class ApplicationStore {
             if (elements.size > 0 && node._elements.some((e) => !elements.has(e))) {
                 return false;
             }
-            return filters.every(([prop, [min, max]]) => {
+            return this.discoveryYearFilter(node) && filters.every(([prop, [min, max]]) => {
```

Comment:
In line 326 there is the optimization whether the filter func is needed at all. In this case it doesn't include the condition whether there is a `discoveryYearFilter`

Reply:
Oh good catch, thank you.

Reply:
Attempted a fix, PTAL.